geodatasets/lib.py

```diff
@@ -118,6 +119,30 @@ def query_name(self, name: str) -> Dataset:
 
         raise ValueError(f"No matching item found for the query '{name}'.")
 
+    def filter_by_geometry(self, geometry_type: str) -> dict[str, Dataset]:
```

Comment:
```suggestion
    def filter_by_geom_type(self, geometry_type: str) -> dict[str, Dataset]:
```
Filter by geometry feels like a spatial query. GeoPandas is using `geom_type` so we can maybe stick to the same?